test/runtime/ct.go

```diff
@@ -0,0 +1,451 @@
+				testCase.src[testCase.kind],
+				identities[testCase.src[helpers.Name]])
+			Expect(err).To(BeNil())
+			CTBefore[testCase.String()] = data
+		}
+
+		for k, v := range CTBefore {
+			Expect(v).To(Equal(6), "CT Tables are not reusing connections for %s", k)
+		}
+
+		res = vm.PolicyDel("id=server-3")
+		res.WasSuccessful()
+
+		By("Checking Policy L7 Dummy")
+		policyL7Dummy := `
+		[{
+			"endpointSelector": {"matchLabels":{"id.server-3":""}},
+			"ingress": [{
+				"fromEndpoints": [{
+				   "matchLabels":{"id.client":""}
+				}],
+				"toPorts": [{
+					"ports": [{"port": "80", "protocol": "tcp"}],
+					"rules": {"http": [{
+						  "path": "/dummy",
+						  "method": "GET"
+					}]}
+				}]
+			}],
+			"labels": ["id=server-3"]
+		}]`
+		_, err = vm.PolicyRenderAndImport(policyL7Dummy)
+		Expect(err).To(BeNil())
+		for _, testCase := range testCombinations {
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.mode, BeFalse)
+		}
+
+		testReach(client, meta[server3][helpers.IPv4], HTTPDummy, BeTrue)
+		testReach(client, meta[server3][helpers.IPv6], HTTPDummy, BeTrue)
+	})
+
+	It("Check the CT cleanup on policyEnforcement default", func() {
+		res := vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
+		res.ExpectSuccess()
+
+		res = vm.PolicyDelAll()
+		res.ExpectSuccess()
+
+		containers(helpers.Create)
+		defer containers(helpers.Delete)
+
+		_, err := vm.PolicyImport(vm.GetFullPath(ctL4Policy), 300)
+		Expect(err).To(BeNil())
+
+		meta := containersMeta()
+		identities, err := vm.GetEndpointsIdentityIds()
+		Expect(err).To(BeNil())
+
+		testCombinations := []connTest{
+			{meta[client], meta[server], helpers.IPv6, http},
+			{meta[client], meta[server], helpers.IPv4, http},
+			{meta[client2], meta[server], helpers.IPv6, http},
+			{meta[client2], meta[server], helpers.IPv4, http},
+		}
+
+		for _, testCase := range testCombinations {
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.mode, BeTrue)
+
+			data, err := countCTEntriesof(
+				testCase.destination[testCase.kind],
+				testCase.src[testCase.kind],
+				identities[testCase.src[helpers.Name]])
+			Expect(err).To(BeNil())
+			Expect(data).To(BeNumerically("<=", 2))
+		}
+		beforeCT, err := vm.ExecCilium("bpf ct list global | wc -l").IntOutput()
+		Expect(err).To(BeNil())
+
+		res = vm.PolicyDel("id=server")
+		res.ExpectSuccess()
+
+		for _, testCase := range testCombinations {
+			data, err := countCTEntriesof(
+				testCase.destination[testCase.kind],
+				testCase.src[testCase.kind],
+				identities[testCase.src[helpers.Name]])
+			Expect(err).To(BeNil())
+			Expect(data).To(Equal(0))
+		}
+
+		afterCT, err := vm.ExecCilium("bpf ct list global | wc -l").IntOutput()
+		Expect(err).To(BeNil())
+		CTDiff := beforeCT - afterCT
+		Expect(CTDiff).To(BeNumerically("<=", 8), "CT map should have exactly 8 entries less and not %d after deleting the policy", CTDiff)
+	})
+})
```

Comment:
It seems the test is not complete, it's lacking all tests after line 422 from `21-ct-clean-up.sh` (`# Install a L4 policy and confirm connectivity`)